### PR TITLE
fix: Diagnostic page issues

### LINKED ISSUE #4858: Duplicate notifications are shown in Diagnostics for Root Bot
I dont have clear repro steps. But we are seeing duplicate notifications in the diagnostics page. U can see it in this bot attached  
https://microsoft.sharepoint.com/:u:/t/ConversationalAI785/EQH0onYA68pFjB1OyMsDAHQBRMIQ1jJ60zYYZ4fQF8M-XQ?e=Ks2OWH

Screen Shot 2020-11-17 at 2 28 17 PM

### PR DESCRIPTION
**Pull request by lei9444**
## Description
1. Duplicate notifications are shown in Diagnostics for Root Bot

> The root cause is the BotIndex.validate contains the checkLUISLocales and checkSkillSetting function. So the diagnostics are created twice.

2. add the link for 'miss skill manifest' warning. will navigate to the design page and open the export skill dialog.

3. separate the diagnostics selector

## Task Item
closes #4858

## Screenshots
diagnostics

### CODE REVIEW

File: Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts [hunk 1]
```diff
@@ -73,34 +70,116 @@ export const diagnosticsSelector = selectorFamily({
...
+});
+
+/**
+ * Check bot skill & setting
+ * 1. used skill not existed in setting
+ * 2. appsettings.json Microsoft App Id or Skill Host Endpoint are empty
+ */
+export const skillDiagnosticsSelectorFamily = selectorFamily({
+  key: 'skillDiagnosticsSelectorFamily',
+  get: (projectId: string) => ({ get }) => {
```
Comment (hatpick) [inline Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts:107]: skillId is better

File: Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts [hunk 2]
```diff
@@ -73,34 +70,116 @@ export const diagnosticsSelector = selectorFamily({
...
+    const botAssets = get(botAssetsSelectFamily(projectId));
+    if (botAssets === null) return [];
+
+    const rootProjectId = get(rootBotProjectIdSelector) ?? projectId;
```
Comment (hatpick) [inline Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts:111]: I'm confused, if this selector is for getting skill diagnostics, why do you care about rootProjectId?

File: Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts [hunk 3]
```diff
@@ -117,16 +211,45 @@ export const diagnosticsSelector = selectorFamily({
...
         const location = `${qnaFile.id}.qna`;
         diagnosticList.push(new QnADiagnostic(rootProjectId, projectId, qnaFile.id, location, diagnostic));
       });
     });
+
     return diagnosticList;
   },
 });
 
+export const diagnosticsSelector = selectorFamily({
```
Comment (hatpick) [inline Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts:240]: diagnosticsSelectorFamily

File: Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts (hunk 1 above)
Comment (lei9444) [reply Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts:107]: Sorry @hatpick the naming confused you here. In fact, this selector family is used to collect the diagnostics about every bot's connected skill settings. I have updated naming

File: Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts (hunk 2 above)
Comment (lei9444) [reply Composer/packages/client/src/recoilModel/selectors/diagnosticsPageSelector.ts:111]: we need to navigate to other pages, This is used to generate the link like
bot/rootid/skill/skillid

**Review by srinaath — APPROVED** [review]

**Pull request merged**